crates/red_knot_python_semantic/src/semantic_index/builder.rs

```diff
@@ -528,6 +532,9 @@ where
     fn visit_stmt(&mut self, stmt: &'ast ast::Stmt) {
         match stmt {
             ast::Stmt::FunctionDef(function_def) => {
+                let definition_inside_loop = self.inside_loop;
+                self.inside_loop = false;
```

Comment:
We also need to unset the flag when entering a lambda expressoin

Reply:
Not saying your suggestion is wrong, just trying to understand: isn't the body of a lambda an expression? or can it contain a (`break`) statement somehow? Or is there another reason why we need to handle it?

Reply:
You're right. Technically it's not possible that the lambda can contain a break. My motivation for adding it to lambda is mainly a precaution in case we use `inside_loop` to detect anything else where this restriction doesn't apply.